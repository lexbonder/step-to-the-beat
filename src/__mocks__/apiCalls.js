/* eslint-disable */
export const getPlaylistData = jest.fn()
  .mockImplementationOnce(() => ({
    tracks: [
      {
        artists: [{name: 'I fight Dragons'}],
        name: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }
    ]
  }))
  .mockImplementationOnce(() => ({
    tracks: [
      {
        artists: [{name: 'I fight Dragons'}],
        name: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }
    ]
  }))
  .mockImplementationOnce(() => ({
    tracks: [
      {
        artists: [{name: 'I fight Dragons'}],
        name: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }
    ]
  }))
  .mockImplementationOnce(() => ({
    tracks: [
      {
        artists: [{name: 'I fight Dragons'}],
        name: 'Dont you?',
        id: '12341234',
        uri: 'spotify:track:12341234'
      }
    ]
  }))

export const getUserName = jest.fn()
  .mockImplementation(() => ({
    display_name: 'Alex Bonder',
    id: 'lxbndr',
    images: [{
      url: 'superlongurl'}
    ]}
  ))

export const createNewPlaylist = jest.fn()
  .mockImplementationOnce(() => ({
    response: `Alex's 148 SPM, ska playlist created successfully`,
    playlistId: '12345abcde'
  }))

export const populatePlaylist = jest.fn()