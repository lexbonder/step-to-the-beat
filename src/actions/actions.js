export const saveAccessToken = token => ({
  type: 'SAVE_ACCESS_TOKEN',
  token
})

export const clearAccessToken = () => ({
  type: 'CLEAR_ACCESS_TOKEN'
})

export const logInUser = () => ({
  type: 'LOG_IN_USER',
  loggedIn: true
})

export const logOutUser = () => ({
  type: 'LOG_OUT_USER',
  loggedIn: false
})

export const saveUserName = userName => ({
  type: 'SAVE_USER_NAME',
  userName
})

export const selectSpm = spm => ({
  type: 'SELECT_SPM',
  spm
})

export const selectGenre = genre => ({
  type: 'SELECT_GENRE',
  genre
})

export const saveSpm = spm => ({
  type: 'SAVE_SPM',
  spm
})