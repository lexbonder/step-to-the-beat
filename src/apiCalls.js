export const getUserName = async (accessToken) => {
  const initialFetch = await fetch('https://api.spotify.com/v1/me', {
      headers:{ Authorization: 'Bearer ' + accessToken }
    })
    const userInfo = await initialFetch.json()
    return userInfo.display_name.split(' ')[0]
}