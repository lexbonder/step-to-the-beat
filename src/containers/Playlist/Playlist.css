.track {
  padding: 20px;
  border-bottom: 2px solid #151515;
}

.song-title {
  color: #ffffff;
  font-size: 25px;
}

.song-artist {
  color: #666666;
  font-size: 20px;
}

.playlist-header {
  border-bottom: 5px solid #404040;
  padding: 10px;
  position: fixed;
  top: 65px;
  width: 100vw;
  background-color: #373737;
}

.playlist-header-top {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.playlist {
  margin-top: 170px;
  padding: 0 3px;
}

.playlist-header h2 {
  color: #63B700;
  padding-left: 10px;
}

#select-all {
  display: inline-block;
}

.playlist-header-bottom,
.track {
  display: grid;
  align-items: center;
  grid-template-columns: 30px auto;

}

.select-all-box {
  padding-left: 10px;
}

.selected {
  background-color: #151515;
  box-shadow: 0 0 5px #63B700;
}
