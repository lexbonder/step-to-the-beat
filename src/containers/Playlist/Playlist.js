import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  removeFromPlaylist,
  addToFavoriteSongs,
  removeFavoriteSong
} from '../../actions/actions';
import './Playlist.css';
import { sendFavSongsToFirebase } from '../../firebaseCalls.js'

export class Playlist extends Component {

  handleDeleteButton = event => {
    const { id } = event.target.parentElement
    this.props.removeFromPlaylist(id)
  }

  handleFavoriteButton = event => {
    const { id } = event.target.parentElement
    const { playlist, favoriteSongs } = this.props
    const song = playlist.find( song => song.id === id)
    const preExisting = favoriteSongs.find(fav => fav.id === song.id)
    if (preExisting) {
      this.props.removeFavoriteSong(song.id)
    } else {
      this.props.addToFavoriteSongs(song)
    }
  }

  componentDidUpdate = () => {
    const { favoriteSongs, user } = this.props
    sendFavSongsToFirebase(user.id, favoriteSongs)
  }

  playlistToRender = () => {
    const { playlist } = this.props;
    return playlist.map(track => (
      <div
        className='track'
        id={track.id}
      >
        <h4>{track.title}</h4>
        <h4>{track.artist}</h4>
        <p onClick={this.handleFavoriteButton}>&#9733;</p>
        <button onClick={this.handleDeleteButton}>Delete</button>
      </div>
    ))
  }

  render() {
    const {spm, genre} = this.props.seeds
    
    return (
      <div>
        <div>
          <h2 className='summary'>
            {`Your ${spm} SPM, ${genre} playlist`}
          </h2>
          <button>Favorite <span>&#9733;</span></button>
          <button>Send to Spotify</button>
        </div>
        <div className='playlist'>
          <h3>Title</h3>
          <h3>Artist</h3>
          <h3>&#9733;</h3>
          <h3>Delete</h3>
          {this.playlistToRender()}
        </div>
      </div>
    )
  }
}

export const MSTP = store => ({
  playlist: store.playlist,
  favoriteSongs: store.favoriteSongs,
  seeds: store.seeds,
  user: store.user
})

export const MDTP = dispatch => ({
  removeFromPlaylist: id => dispatch(removeFromPlaylist(id)),
  addToFavoriteSongs: song => dispatch(addToFavoriteSongs(song)),
  removeFavoriteSong: id => dispatch(removeFavoriteSong(id))
})

export default withRouter(connect(MSTP, MDTP)(Playlist))