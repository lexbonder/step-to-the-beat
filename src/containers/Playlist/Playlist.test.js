/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { savePlaylist } from '../../actions/actions';
import { Playlist, MSTP, MDTP } from './Playlist';
import * as apiCalls from '../../apiCalls';

jest.mock('../../apiCalls')

describe('Playlist', () => {
  let wrapper;
  let mockPlaylist = [{
    artist: 'I Fight Dragons',
    title: 'Dont You?',
    id: '1234abcd',
    uri: 'spotify:tracks:1234abcd'
  }]
  let mockNewSeed = {
    spm: 148,
    genre: 'ska'
  }
  let mockUser = {
    name: 'Alex',
    id: 'lxbndr'
  }
  let mockAccessToken = '12345abcde'

  beforeEach( ()=> {
    wrapper = shallow(<Playlist 
      playlist={mockPlaylist}
      newSeed={mockNewSeed}
      user={mockUser}
      accessToken={mockAccessToken}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {
    it('should set the playlistName and trackUris into state on mount', () => {
      const mockState = {
        playlistName: 'Alex\'s 148 SPM, ska playlist',
        errorStatus: '',
        page: 2,
        selectedTracks: [],
      }
      expect(wrapper.state()).toEqual(mockState)
    })
  })

  describe('playlistToRender', () => {
    it('should match the snapshot', () => {
      expect(wrapper.instance().playlistToRender).toMatchSnapshot()
    })
  })

  describe('toggleSelectAll', () => {
    it('should match the snapshot when event.target.checked is true', () => {
      const mockEvent = {target: {checked: true}}

      expect(wrapper.instance().toggleSelectAll(mockEvent)).toMatchSnapshot()
    })

    it('should match the snapshot when event.target.checked is false', () => {
      const mockEvent = {target: {checked: false}}

      expect(wrapper.instance().toggleSelectAll(mockEvent)).toMatchSnapshot()
    })
  })

  describe('sendToSpotify', () => {
    it('should match the snapshot if the playlist is created in spotify', () => {
      expect(wrapper.instance().sendToSpotify()).toMatchSnapshot()
    })

    it('should call createNewPlaylist with a user ID, accessToken, and a playlist name', () => {
      wrapper.instance().sendToSpotify()
      expect(apiCalls.createNewPlaylist).toHaveBeenCalledWith('lxbndr', '12345abcde', 'Alex\'s 148 SPM, ska playlist')
    })

    it('should call populatePlaylist with a user ID, plylist ID, accessToken, and selectedSongs', () => {
      wrapper.instance().sendToSpotify()
      expect(apiCalls.populatePlaylist).toHaveBeenCalledWith('lxbndr', '12345abcde', '12345abcde', [])
    })

    it('should set the state of errorStatus to display an error if playlist creation fails', async () => {
      await wrapper.instance().sendToSpotify()
      expect(wrapper.state().errorStatus).not.toEqual('')
    })
  })

  // describe('changePlaylistName', () => {
  //   it('should set state of playlistName with whatever is in the input', () => {
  //     const mockEvent = {target: {value: 'Rock n Roll Son!'}}
  //     wrapper.instance().changePlaylistName(mockEvent)
  //     expect(wrapper.state().playlistName).toEqual('Rock n Roll Son!')
  //   })
  // })

  describe('getMoreSongs', () => {
    beforeEach(() => {
      // Mock user action of scrolling to bottom of page
      window.innerHeight = 0;
    })

    it('should call getPlaylistData with a spm, genre, accessToken, and limit of 40', async () => {
      await wrapper.instance().getMoreSongs()
      expect(apiCalls.getPlaylistData).toHaveBeenCalledWith(148, 'ska', '12345abcde', 40)
    })

    it('should call getPlaylistData with a spm, genre, accessToken, and limit of 60 when called twice', async () => {
      await wrapper.instance().getMoreSongs()
      await wrapper.instance().getMoreSongs()
      expect(apiCalls.getPlaylistData).toHaveBeenCalledWith(148, 'ska', '12345abcde', 60)
    })

    it('should increment the page in state', async () => {
      expect(wrapper.state().page).toEqual(2)
      await wrapper.instance().getMoreSongs()
      expect(wrapper.state().page).toEqual(3)
    })
  })

  describe('MSTP', () => {
    it('should return an object with values matching whatever is in state', () => {
      const mockState = {
        playlist: mockPlaylist,
        newSeed: mockNewSeed,
        user: mockUser,
        accessToken: mockAccessToken
      }
      const mapped = MSTP(mockState);

      expect(mapped.playlist).toEqual(mockPlaylist)
      expect(mapped.newSeed).toEqual(mockNewSeed)
      expect(mapped.user).toEqual(mockUser)
      expect(mapped.accessToken).toEqual(mockAccessToken)
    })
  })

  describe('MDTP', () => {
    it('should call dispatch when addToPlaylist is called', () => {
      const mockPlaylist = [{song: 'name'}]
      const mockDispatch = jest.fn();
      const result = MDTP(mockDispatch);
      result.savePlaylist(mockPlaylist);
      expect(mockDispatch).toHaveBeenCalledWith(savePlaylist(mockPlaylist))
    })
  })
})