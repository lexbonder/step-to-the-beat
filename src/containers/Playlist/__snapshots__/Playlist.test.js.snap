// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist playlistToRender should match the snapshot 1`] = `[Function]`;

exports[`Playlist sendToSpotify should match the snapshot if the playlist is created in spotify 1`] = `Promise {}`;

exports[`Playlist should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Playlist
    accessToken="12345abcde"
    newSeed={
        Object {
            "genre": "ska",
            "spm": 148,
          }
    }
    playlist={
        Array [
            Object {
              "artist": "I Fight Dragons",
              "id": "1234abcd",
              "title": "Dont You?",
              "uri": "spotify:tracks:1234abcd",
            },
          ]
    }
    user={
        Object {
            "id": "lxbndr",
            "name": "Alex",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="playlist-header"
>
          <div
                    className="playlist-header-top"
          >
                    <h2>
                              Alex's 148 SPM, ska playlist
                    </h2>
                    <button
                              className="buttons send"
                              disabled={true}
                              onClick={[Function]}
                    >
                              Send to Spotify
                    </button>
          </div>
          <div
                    className="playlist-header-bottom"
          >
                    <div
                              className="select-all-box"
                    >
                              <p>
                                        All
                              </p>
                              <input
                                        id="select-all"
                                        onClick={[Function]}
                                        type="checkbox"
                              />
                    </div>
                    <h2>
                              0 of 1
                                            selected (max 100)
                    </h2>
          </div>
</div>,
        <div
          className="playlist"
>
          <div
                    className="
                    track 
                    "
                    id="1234abcd"
                    onClick={[Function]}
          >
                    <input
                              className="checkbox"
                              id="spotify:tracks:1234abcd"
                              onClick={[Function]}
                              type="checkbox"
                    />
                    <div>
                              <h4
                                        className="song-title"
                              >
                                        Dont You?
                              </h4>
                              <h4
                                        className="song-artist"
                              >
                                        I Fight Dragons
                              </h4>
                    </div>
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="playlist-header-top"
>
              <h2>
                            Alex's 148 SPM, ska playlist
              </h2>
              <button
                            className="buttons send"
                            disabled={true}
                            onClick={[Function]}
              >
                            Send to Spotify
              </button>
</div>,
            <div
              className="playlist-header-bottom"
>
              <div
                            className="select-all-box"
              >
                            <p>
                                          All
                            </p>
                            <input
                                          id="select-all"
                                          onClick={[Function]}
                                          type="checkbox"
                            />
              </div>
              <h2>
                            0 of 1
                                          selected (max 100)
              </h2>
</div>,
          ],
          "className": "playlist-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Alex's 148 SPM, ska playlist
</h2>,
                <button
                  className="buttons send"
                  disabled={true}
                  onClick={[Function]}
>
                  Send to Spotify
</button>,
              ],
              "className": "playlist-header-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alex's 148 SPM, ska playlist",
                },
                "ref": null,
                "rendered": "Alex's 148 SPM, ska playlist",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Send to Spotify",
                  "className": "buttons send",
                  "disabled": true,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Send to Spotify",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="select-all-box"
>
                  <p>
                                    All
                  </p>
                  <input
                                    id="select-all"
                                    onClick={[Function]}
                                    type="checkbox"
                  />
</div>,
                <h2>
                  0 of 1
                                selected (max 100)
</h2>,
              ],
              "className": "playlist-header-bottom",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      All
</p>,
                    <input
                      id="select-all"
                      onClick={[Function]}
                      type="checkbox"
/>,
                  ],
                  "className": "select-all-box",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "All",
                    },
                    "ref": null,
                    "rendered": "All",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "id": "select-all",
                      "onClick": [Function],
                      "type": "checkbox",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0 of 1
              selected (max 100)",
                },
                "ref": null,
                "rendered": "0 of 1
              selected (max 100)",
                "type": "h2",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="
          track 
          "
              id="1234abcd"
              onClick={[Function]}
>
              <input
                            className="checkbox"
                            id="spotify:tracks:1234abcd"
                            onClick={[Function]}
                            type="checkbox"
              />
              <div>
                            <h4
                                          className="song-title"
                            >
                                          Dont You?
                            </h4>
                            <h4
                                          className="song-artist"
                            >
                                          I Fight Dragons
                            </h4>
              </div>
</div>,
          ],
          "className": "playlist",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1234abcd",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="checkbox"
                  id="spotify:tracks:1234abcd"
                  onClick={[Function]}
                  type="checkbox"
/>,
                <div>
                  <h4
                                    className="song-title"
                  >
                                    Dont You?
                  </h4>
                  <h4
                                    className="song-artist"
                  >
                                    I Fight Dragons
                  </h4>
</div>,
              ],
              "className": "
          track 
          ",
              "id": "1234abcd",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "checkbox",
                  "id": "spotify:tracks:1234abcd",
                  "onClick": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="song-title"
>
                      Dont You?
</h4>,
                    <h4
                      className="song-artist"
>
                      I Fight Dragons
</h4>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Dont You?",
                      "className": "song-title",
                    },
                    "ref": null,
                    "rendered": "Dont You?",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "I Fight Dragons",
                      "className": "song-artist",
                    },
                    "ref": null,
                    "rendered": "I Fight Dragons",
                    "type": "h4",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="playlist-header"
>
            <div
                        className="playlist-header-top"
            >
                        <h2>
                                    Alex's 148 SPM, ska playlist
                        </h2>
                        <button
                                    className="buttons send"
                                    disabled={true}
                                    onClick={[Function]}
                        >
                                    Send to Spotify
                        </button>
            </div>
            <div
                        className="playlist-header-bottom"
            >
                        <div
                                    className="select-all-box"
                        >
                                    <p>
                                                All
                                    </p>
                                    <input
                                                id="select-all"
                                                onClick={[Function]}
                                                type="checkbox"
                                    />
                        </div>
                        <h2>
                                    0 of 1
                                                  selected (max 100)
                        </h2>
            </div>
</div>,
          <div
            className="playlist"
>
            <div
                        className="
                      track 
                      "
                        id="1234abcd"
                        onClick={[Function]}
            >
                        <input
                                    className="checkbox"
                                    id="spotify:tracks:1234abcd"
                                    onClick={[Function]}
                                    type="checkbox"
                        />
                        <div>
                                    <h4
                                                className="song-title"
                                    >
                                                Dont You?
                                    </h4>
                                    <h4
                                                className="song-artist"
                                    >
                                                I Fight Dragons
                                    </h4>
                        </div>
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="playlist-header-top"
>
                <h2>
                                Alex's 148 SPM, ska playlist
                </h2>
                <button
                                className="buttons send"
                                disabled={true}
                                onClick={[Function]}
                >
                                Send to Spotify
                </button>
</div>,
              <div
                className="playlist-header-bottom"
>
                <div
                                className="select-all-box"
                >
                                <p>
                                                All
                                </p>
                                <input
                                                id="select-all"
                                                onClick={[Function]}
                                                type="checkbox"
                                />
                </div>
                <h2>
                                0 of 1
                                              selected (max 100)
                </h2>
</div>,
            ],
            "className": "playlist-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Alex's 148 SPM, ska playlist
</h2>,
                  <button
                    className="buttons send"
                    disabled={true}
                    onClick={[Function]}
>
                    Send to Spotify
</button>,
                ],
                "className": "playlist-header-top",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Alex's 148 SPM, ska playlist",
                  },
                  "ref": null,
                  "rendered": "Alex's 148 SPM, ska playlist",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Send to Spotify",
                    "className": "buttons send",
                    "disabled": true,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Send to Spotify",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="select-all-box"
>
                    <p>
                                        All
                    </p>
                    <input
                                        id="select-all"
                                        onClick={[Function]}
                                        type="checkbox"
                    />
</div>,
                  <h2>
                    0 of 1
                                  selected (max 100)
</h2>,
                ],
                "className": "playlist-header-bottom",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p>
                        All
</p>,
                      <input
                        id="select-all"
                        onClick={[Function]}
                        type="checkbox"
/>,
                    ],
                    "className": "select-all-box",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "All",
                      },
                      "ref": null,
                      "rendered": "All",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "id": "select-all",
                        "onClick": [Function],
                        "type": "checkbox",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0 of 1
              selected (max 100)",
                  },
                  "ref": null,
                  "rendered": "0 of 1
              selected (max 100)",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="
          track 
          "
                id="1234abcd"
                onClick={[Function]}
>
                <input
                                className="checkbox"
                                id="spotify:tracks:1234abcd"
                                onClick={[Function]}
                                type="checkbox"
                />
                <div>
                                <h4
                                                className="song-title"
                                >
                                                Dont You?
                                </h4>
                                <h4
                                                className="song-artist"
                                >
                                                I Fight Dragons
                                </h4>
                </div>
</div>,
            ],
            "className": "playlist",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1234abcd",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="checkbox"
                    id="spotify:tracks:1234abcd"
                    onClick={[Function]}
                    type="checkbox"
/>,
                  <div>
                    <h4
                                        className="song-title"
                    >
                                        Dont You?
                    </h4>
                    <h4
                                        className="song-artist"
                    >
                                        I Fight Dragons
                    </h4>
</div>,
                ],
                "className": "
          track 
          ",
                "id": "1234abcd",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "checkbox",
                    "id": "spotify:tracks:1234abcd",
                    "onClick": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="song-title"
>
                        Dont You?
</h4>,
                      <h4
                        className="song-artist"
>
                        I Fight Dragons
</h4>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Dont You?",
                        "className": "song-title",
                      },
                      "ref": null,
                      "rendered": "Dont You?",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "I Fight Dragons",
                        "className": "song-artist",
                      },
                      "ref": null,
                      "rendered": "I Fight Dragons",
                      "type": "h4",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Playlist toggleSelectAll should match the snapshot when event.target.checked is false 1`] = `undefined`;

exports[`Playlist toggleSelectAll should match the snapshot when event.target.checked is true 1`] = `undefined`;
